packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/index.tsx

```diff
@@ -36,6 +42,52 @@ const Th = ({ className, children }: CellProps) => (
   </th>
 );
 
+function ClampedText({ text }: { text: string }) {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [hasOverflow, setHasOverflow] = useState(false);
+  const [hasBigword, setHasBigword] = useState(false);
+  const textRef = useRef<HTMLDivElement>(null);
+
+  useLayoutEffect(() => {
+    if (textRef.current) {
+      const numLines = textRef.current?.getBoundingClientRect().height / 16;
+      setHasOverflow(numLines >= 4);
+    }
+    setHasBigword(text.split(" ").some((word) => word.length > 20));
+  }, []);
+
+  const toggleExpanded = () => {
+    setIsExpanded(!isExpanded);
+  };
+
+  return (
+    <div>
+      <div
+        ref={textRef}
+        className={cn(
+          "break-all",
+          !isExpanded && hasBigword
+            ? "overflow-hidden text-ellipsis line-clamp-3"
+            : "",
+          !isExpanded && hasOverflow ? "line-clamp-3" : ""
```

Comment:
classnames tip 😛 
```suggestion
          {
            "overflow-hidden text-ellipsis": !isExpanded && hasBigword,
            "line-clamp-3": hasBigWord || hasOverflow,
          }
```
but couldn't you simply always have `break-all line-clamp-3 text-ellipsis`? It sounds like that would achieve the same result you want -- regardless of any additional logic?